scripts/configure_embeddings.py

```
@@ -0,0 +1,438 @@
+        target_dim=target_dim,
+        schema=schema,
+        current_dims=current,
+        needs_alter=needs_alter,
+    )
+
+
+async def _count_non_null_embeddings(
+    conn: AsyncConnection, schema: str, table: str
+) -> int:
+    query = text(
+        f'SELECT COUNT(*) AS n FROM "{schema}"."{table}" WHERE embedding IS NOT NULL'
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)